typedb/connection/client.py

```diff
@@ -32,23 +36,30 @@
 
 
 class _TypeDBClientImpl(TypeDBClient):
+    _PULSE_INTERVAL_SECONDS = 5
 
     # TODO: Detect number of available CPUs
     def __init__(self, address: str, parallelisation: int = 2):
         self._address = address
         self._transmitter = RequestTransmitter(parallelisation)
         self._sessions: Dict[bytes, _TypeDBSessionImpl] = {}
+        self._sessions_lock = Lock()
```

Comment:
We'd previously believed that `dict` was thread-safe in Python, but this has now been disproven - so we need to restrict access to our `sessions` collection using a Lock.